README.md

```diff
@@ -87,86 +87,69 @@ To learn more about the different security schemes please the check the [officia
-      }
-    }
-  end
+  @doc """
+  List users
+  """
+  @doc responses: %{
+    200 => {"Users", "application/json", MyAppWeb.Schema.Users}
+  }
+  def index(conn, _params) do
+    {:ok, users} = MyApp.Users.all()
 
-  # Controller's `show` action
-  def show(conn, %{id: id}) do
-    {:ok, user} = MyApp.Users.find_by_id(id)
-    json(conn, 200, user)
+    json(conn, users)
   end
 
+  @doc """
+  Update user
+  """
+  @doc parameters: [
+      id: [in: :query, type: :string, required: true, description: "User ID"]
+    ],
+    request_body: {"Request body to update a User", "application/json", User, required: true},
+    responses: %{
+      200 => {"User", "application/json", MyAppWeb.Schema.User}
+    }
+  def update(conn, %{id: id}) do
+    with {:ok, user} <- MyApp.Users.update(conn.body_params) do
+      json(conn, user)
+    end
+  end
 end
 ```
 
-Alternatively, you can create an operation file separately using `defdelegate`.
+The responses can also be defined using keyword list syntax,
+and the HTTP status codes can be replaced with their text equivalents:
 
 ```elixir
-# Phoenix's controller
-defmodule MyAppWeb.UserController do
-  defdelegate open_api_operation(action), to: MyAppWeb.UserApiOperation
-
-  def show(conn, %{id: id}) do
-    {:ok, user} = MyApp.Users.find_by_id(id)
-    json(conn, 200, user)
-  end
-end
+  @doc responses: [
+    ok: {"User", "application/json", MyAppWeb.Schema.User}
+    unprocessible_entity: {"Bad request parameters", "application/json", MyAppWeb.Schema.BadRequestParameters},
+    not_found: {"Not found", "application/json", MyAppWeb.Schema.NotFound}
+  ]
+```
 
-# Open API Spex operations
-defmodule MyAppWeb.UserApiOperation do
-  alias OpenApiSpex.Operation
-  alias MyAppWeb.Schemas.UserResponse
+The full set of atom keys are defined in `Plug.Conn.Status.code/1`.
 
-  @spec open_api_operation(atom) :: Operation.t()
-  def open_api_operation(action) do
-    operation = String.to_existing_atom("#{action}_operation")
-    apply(__MODULE__, operation, [])
-  end
+Each definition in a controller action or plug operation is converted
+to an `%OpenApiSpex.Operation{}` struct. The definitions are read
+by your application's `ApiSpec` module, which in turn is
+called from the `OpenApiSpex.Plug.PutApiSpex` plug on each request.
+The definitions data is cached, so it does actually extract the definitions on each request.
 
-  @spec show_operation() :: Operation.t()
-  def show_operation() do
-    %Operation{
-      tags: ["users"],
-      summary: "Show user",
-      description: "Show a user by ID",
-      operationId: "UserController.show",
-      parameters: [
-        Operation.parameter(:id, :path, :integer, "User ID", example: 123, required: true)
-      ],
-      responses: %{
-        200 => Operation.response("User", "application/json", UserResponse)
-      }
-    }
-  end
-end
-```
+If the ExDoc-based operation specs don't provide the flexibiliy you need, the `%Operation{}` struct
```

Comment:
Is the difference in key naming conventions called out somewhere? The `Operation` and `Schema` structs use camel case to match OpenAPI / JSON Schema, whereas the @doc attributes use snake case.

Reply:
Good point. I'll add that.